src/spec-node/singleContainer.ts

```diff
@@ -159,15 +178,34 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 	}
 
 	const args: string[] = [];
-	if (buildParams.useBuildKit) {
+	if (buildParams.useBuildKit && !buildParams.enableBuildx) {
 		args.push('buildx', 'build',
 			'--load', // (short for --output=docker, i.e. load into normal 'docker images' collection)
 			'--build-arg', 'BUILDKIT_INLINE_CACHE=1', // ensure cache manifest is included in the image
 		);
+	} else if (buildParams.enableBuildx && argImageName) {
```

Comment:
My feeling here is that this would be better merged with the above branch, i.e. have a single branch that handles the BuildKit/buildx path and optionally adds the `--platform` args as needed

Reply:
I agree, we no longer need a `--buildx` flag. We also `--load` by default, so maybe we only need `--platform` and `--push` then?

Reply:
Currently, when building, the CLI generates an image name based on the devcontainer folder etc and uses that to build. If the `--image-name` option is passed then the image is tagged with that name after the build.

So if the caller adds a `docker push` call after calling the `devcontainer` CLI then that will work. To add the `--push` option to the `devcontainer` CLI, I think we would need to change the CLI behaviour to use the name specified via `--image-name` so that the name of the image being built (and pushed) is correct?

Reply:
My understanding is that `--push` (when passed to `buildx build`) would push at the end of the build to avoid `--load` because `--load` only works for the local platform. So we would have to use the correct tag on `buildx build`. (I just noticed that we can use multiple tags on the same build if we have to.)

Reply:
Yeah, ignore my previous comment. You're right, the `--load` will drop the multi-platform aspect of the image!

I'm not sure what the behaviour would be if we have multiple tags when building and also include the `--push` option, does it attempt to push all tags?

We could modify the `buildNamedImageAndExtend` function to take an `imageName` and then use that if it's set in preference of the [other options currently used](https://github.com/devcontainers/cli/blob/b6ee1258c203672be53b669183f2a3c35ed3da5f/src/spec-node/singleContainer.ts#L105)? This would save the later [`docker tag` step](https://github.com/devcontainers/cli/blob/main/src/spec-node/devContainersSpecCLI.ts#L331)

Reply:
Hi @stuartleeks, @chrmarti . These are great comments. Thanks so much. With regards the comment about `--load`, it is not possible to build more than one image platform at the time for instance, so kind of useful for knowing that the image build works for a specific platform.
Also `--push` is quite useful to just push to a container registry with all the platforms specified (>= 1) without the need to create one at the time and keep track of the manifest that encapsulates all the images. (one manifest uploaded with all the images references too), which is what this enhancement aims for.

Reply:
Sounds good, so we only need `--platform` and `--push`. I think without `--push` we would still default to `--load` and potentially having multiple tags is fine. With `--push` we can omit the automatically generated tag and only use the user-specified one (unless I'm missing something).

Reply:
Is there any reason not to pass the value of `--image-name` into the `buildImageAndExtend` function, defaulting to the current generated name if no value is specified? Then we could just apply a single tag, and skip the `docker tag` step in the CLI file

Reply:
> Is there any reason not to pass the value of `--image-name` into the `buildImageAndExtend` function, defaulting to the current generated name if no value is specified? Then we could just apply a single tag, and skip the `docker tag` step in the CLI file

@chrmarti, your comments sounds reasonable. For the image name, it is needed for `--push` since it is the container registry name and not the autogenerated name. I'll make the changes. Thanks to both of you for all the comments.

Reply:
@juzuluag I agree that the image name specified in the command args needs to be passed through.

My question above was around whether there is any reason that in this case we would also need the generated image name? Or whether, when the user has specified the image name, it would be sufficient to just apply that one as a tag.

If so, I think it simplifies the code around specifying the tags as it no longer needs to account for whether `--push` has been specified.

Reply:
Hi @chrmarti  and @stuartleeks . I made some changes, so let me know if these are more aligned with your thoughts.

Reply:
hmm, I would defer to @chrmarti . I just did some changes, so hopefully that gives us more insights. Thanks